### ISSUE TITLE
Skip "publishable" check when only example build files are edited

### ISSUE THREAD

**Issue by jmagman** [flutter/flutter#105181]
Pass "publishable" check in plugins and packages when the only changed files are in the example directories.

https://github.com/flutter/plugins/blob/main/script/tool/lib/src/publish_check_command.dart

This will allow us to autosubmit by putting `waiting for tree` labels on roller PRs that update examples like https://github.com/flutter/plugins/pull/5858.

**Comment by stuartmorgan** [comment]
I reworked the way overriding the `publishable` check works (it's now just adding a label and re-running `publishable`, which is very fast), and also improved the logic for test-only Dependabot changes. Given that it's now easy to override, I think it's useful to still have a human decide if an example update is worth mentioning in the CHANGELOG or not (e.g., sometimes examples are showing a new recommendation, or fixing something that was shown but we've decided was a bad idea, in which case calling it out is helpful), so I'm going to narrow the focus here.

We can have the tooling exempt any build-only changes (e.g., build.gradle) in `example/`, since they are unlikely to ever be interesting to mention.
